pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -665,6 +665,7 @@ func buildSidecarOutboundHTTPListenerOptsForPortOrUDS(listenerMapKey *string,
 
 		protocol: listenerOpts.port.Protocol,
 		class:    istionetworking.ListenerClassSidecarOutbound,
+		svc:      listenerOpts.service,
```

Comment:
how can an http listener have a service? they are shared. by all services in the same port

Reply:
Sorry. I should have put this in draft. I meant to use [StatefulSessionPerRoute](https://github.com/envoyproxy/envoy/blob/83ca9cf6d304de5e5cd9b44e95aab55cfcd4e735/api/envoy/extensions/filters/http/stateful_session/v3/stateful_session.proto#L28 per route) not at the connection manager level.

Reply:
Envoy implementation requires the filter to be set at listener level - it can be set with an invalid cookie path ( so it doesn't generate a cookie), but it must be set.

I think we can require that services using sessions have dedicated ports.

Reply:
We can set it at route level with actual config. The top level one at listener can be with empty config and does not have any impact